pagopa/io-app
# feat(IT Wallet): [SIW-1295] Add eID issuance flow with SPID and CieID

> [!WARNING]
> This PR depends on #5934

## Short description
This PR introduces the eID issuance flow with the following identification methods:
- SPID
- CieID

**Note:** the eID is persisted using `react-native-async-storage`. This is a temporary solution; the credential will be persisted differently in a future task/PR.

## List of changes proposed in this pull request
- eID issuance machine updated
- Moved getting the wallet attestation to the eID issuing actor for simplicity
- `itwIssuanceUtils` added
- Added failure screen to handle the most common error cases

## How to test
> [!CAUTION]
> The changes were tested extensively on Android, but I was unable to do the same on iOS as I can not use my personal account with IO bundle ID. 

The eID issuance was tested against the remote `io-wallet` functions, with the local Python dev server found in `io-react-native-wallet` example folder. Additionally, a proxy was used to inject an authorization header when calling IPZS services.

Since this PR includes SIW-1296, you need to make sure the authenticated user is the same that is requesting the eID. For testing purposes, build the app with the env flag `ITW_BYPASS_IDENTITY_MATCH` set to `YES`.

https://github.com/user-attachments/assets/51f0bc0e-bbae-4402-b024-34c36f882124
